Applications/SimpleShot/task_runner.cpp

```diff
@@ -175,15 +233,161 @@ std::unique_ptr<ml::train::Model> createModel(const std::string &backbone,
 }
 } // namespace simpleshot
 
+/**
+ * @brief Check whether file exist or not
+ * @param file_name name of file
+ * @return bool
+ */
+
+bool isFileExist(std::string file_name) {
+  std::ifstream infile(file_name);
+  return infile.good();
+}
+
+/**
+ * @brief appends label in allLabels.txt file that maintains all added classes
+ * string labels
+ * @param label_path string label path of newly added classes
+ * @return int
+ */
+
+int append_label(std::string label_path) {
+
+  int class_cnt = 0;
+  int flag = 0;
+  std::map<int, std::string> label_map;
+
+  if (std::strcmp(label_path.c_str(), "null") != 0) {
+
+    std::fstream label_file;
+    label_file.open(label_path, std::ios::in);
+
+    std::cout << "Name of new classes which is being added:\n";
+
+    if (label_file.is_open()) { /// checking whether the file is open
+      std::string tp;
+
+      while (std::getline(label_file, tp)) {
+        /// read data from file object and put it into string
+        /// print the data of the string
+        std::cout << tp << "\n";
+
+        label_map.insert(std::pair<int, std::string>(class_cnt, tp));
+        class_cnt++;
+      }
+      label_file.close();
+    }
+
+    simpleshot::new_classes = class_cnt;
+
+    std::cout << "\n Number of new classes added : " << class_cnt << "\n \n";
+
+  }
+
+  else {
+    flag = 1;
+  }
+
+  /// Read Create the allLabels.txt
+
+  std::string all_class_label_path = "allLabels.txt";
+
+  std::fstream all_labels_file;
+
+  if (isFileExist(all_class_label_path) == false) {
+
+    simpleshot::earlier_classes = 0;
+
+    all_labels_file.open(all_class_label_path, std::ios::out);
+
+    /// checking whether the file is open
+
+    if (all_labels_file.is_open()) {
+
+      for (auto itr = label_map.begin(); itr != label_map.end(); itr++) {
+        all_labels_file << itr->second;
+        all_labels_file << "\n";
+      }
+
+      /// close the file object
+
+      all_labels_file.close();
+    }
+
+  }
+
+  else {
+    std::map<int, std::string> earlier_label_map;
+
+    all_labels_file.open(all_class_label_path, std::ios::in);
+    std::cout << "Name of Earlier Classes which were present: \n";
+    class_cnt = 0;
+
+    if (all_labels_file.is_open()) {
+      /
```

Comment:
this extra "/" is causing build errors on ubuntu and tizen.

Reply:
Ok , I will remove it, it's by mistake I will fix that.

Reply:
Please Check now